### PR TITLE
Wire up admin GenericOidcClientProvider

### PR DESCRIPTION
**Pull request by yotommy**
### Description

This change wires up the new admin `GenericOidcClientProvider` to the `SecurityModule`.

It also registers new config variables to select and support the new provider. However, the values of the config variables are not yet accessed through `SettingsManifest`; instead, they are accessed directly from the `Config`. When the applicant and admin versions are refactored to have a common superclass, they will be revised to use the getters from `SettingsManifest`.

Tested with Okta on a dev instance.

#### General

Read the full guidelines for PRs [here](https://docs.civiform.us/contributor-guide/developer-guide/technical-contribution-guide#guidelines-for-pull-requests)

- [x] Added the correct label: < feature | bug | dependencies | infrastructure | ignore-for-release | database >
- [ ] Created unit and/or browser tests which fail without the change (if possible)
- [x] Performed manual testing (Chrome and Firefox if it includes front-end changes)
- [ ] Extended the README / documentation, if necessary

### Issue(s) this completes

Relates to #5401

### CODE REVIEW

**Comment by yotommy** [comment]
Current dependencies on/for this PR:
* main
  * **PR #5415** <a href="https://app.graphite.dev/github/pr/civiform/civiform/5415" target="_blank">Graphite</a> 
    * **PR #5416** <a href="https://app.graphite.dev/github/pr/civiform/civiform/5416" target="_blank">Graphite</a> 
      * **PR #5417** <a href="https://app.graphite.dev/github/pr/civiform/civiform/5417" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/civiform/civiform/5417?utm_source=stack-comment).

File: server/app/auth/AuthIdentityProviderName.java [hunk 1]
```diff
@@ -44,6 +46,28 @@ public static AuthIdentityProviderName fromConfig(Config config) {
...
+    }
+    String supportedOptions =
+        Arrays.stream(AuthIdentityProviderName.values())
+            .map(AuthIdentityProviderName::getValue)
+            .collect(Collectors.joining(", "));
+    throw new IllegalArgumentException(
+        "Unsupported civiform_admin_idp value: "
+            + providerName
+            + ". Supported values are "
+            + supportedOptions);
```
Comment (bion) [inline server/app/auth/AuthIdentityProviderName.java:68]: **suggestion (blocking):** could this be replaced with:
```suggestion
    String providerName = config.getString(AUTH_ADMIN_CONFIG_PATH);
    try {
      return AuthIdentityProviderName.valueOf(providerName);
    } catch (IllegalArgumentException e) {
      String supportedOptions =
          Arrays.stream(AuthIdentityProviderName.values())
              .map(AuthIdentityProviderName::getValue)
              .collect(Collectors.joining(", "));
      throw new IllegalArgumentException(
          "Unsupported civiform_admin_idp value: "
              + providerName
              + ". Supported values are "
              + supportedOptions);
    }
```

File: server/app/modules/SecurityModule.java [hunk 2]
```diff
@@ -120,20 +120,39 @@ protected void configure() {
...
+              .annotatedWith(AdminAuthClient.class)
+              .toProvider(auth.oidc.admin.GenericOidcClientProvider.class);
+          logger.info("Using Generic OIDC for admin auth provider");
+          break;
+        default:
+          throw new ConfigurationException("Unable to create admin identity provider: " + idpName);
+      }
+    } catch (RuntimeException e) {
+      logger.error("Error getting admin auth provider");
+      throw e;
```
Comment (bion) [inline server/app/modules/SecurityModule.java:149]: **question (blocking):** when do we expect a RuntimeException with this code? Is the `try/catch` necessary here?

**Review by bion — CHANGES_REQUESTED** [review]

File: server/app/auth/AuthIdentityProviderName.java (hunk 1 above)
Comment (yotommy) [reply server/app/auth/AuthIdentityProviderName.java:68]: The parameter supplied to `.valueOf()` must be the name (e.g., `"GENERIC_OIDC_ADMIN"`), not the associated value (e.g., `"generic-oidc-admin"`). But it looks so much nicer your way that I created a helper function to mimic the idea.

**Review by bion — APPROVED** [review]

**Pull request merged**